### DISCUSSION TITLE
majuscule en début de déclaration de type

### DISCUSSION THREAD

**Discussion by gorky1234** [moshell-lang/moshell#121] [Ideas]
Bonjour,

Je trouve que mettre une majuscule en début de déclaration de type de fait perdre en productivités.
Pouvez-vous corriger ça, s'il vous plaît ?

Cordialement

**Comment by syldium** [comment]
You can compile with the following command to remove the uppercase letter:
```bash
sed -i 's/"[A-Z]/\L&/' analyzer/src/types/ctx.rs && cargo build --release
```

**Comment by gorky1234** [comment]
Ce n'est pas très pratique, pourquoi ne pas l'intégrer directement dans le code source pour que les autres développeurs voulant être le plus productif possible?

**Comment by Override-6** [comment]
Because we want a naming convention that distinguishes members (variables, functions) and types (generics / class).
You may loose a millisecond to push the shift button, but you'll earn more time not loosing it to try to figure out the types from the members.
Also a code with the same naming types for every components isn't really clear

**Comment by gorky1234** [comment]
I don't agree, but if you want
